splunk-otel-android/src/main/java/com/splunk/rum/ActivityTracerCache.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import android.app.Activity;
+import androidx.annotation.VisibleForTesting;
+import io.opentelemetry.api.trace.Tracer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+/**
+ * Encapsulates the fact that we have an ActivityTracer instance per Activity class, and provides
+ * convenience methods for adding events and starting spans.
+ */
+public class ActivityTracerCache {
```

Comment:
Nit: probably shouldn't be public
```suggestion
class ActivityTracerCache {
```

Reply:
not accepting suggestion because I also got some `public` methods. 👍🏻